src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_AssignmentOperator.cs

```diff
@@ -301,7 +301,7 @@ private BoundExpression VisitAssignmentOperator(BoundAssignmentOperator node, bo
             if (setMethod is null)
             {
                 var autoProp = (SourcePropertySymbolBase)property.OriginalDefinition;
-                Debug.Assert(autoProp.IsAutoPropertyWithGetAccessor,
+                Debug.Assert(autoProp.IsAutoProperty && autoProp.GetMethod is not null,
```

Comment:
>autoProp.IsAutoProperty && autoProp.GetMethod is not null [](http://example.com/codeflow?start=29&length=57)

I think you are missing the point of the activity that we are doing. A simple in-lining of implementation isn't the kind of refactoring I had in mind. Think what is the purpose of this check, i.e. what are we trying actually to check. Then think if it is worth adding a new property that has that specific purpose in mind. If you cannot come up with anything, skip the usage.